### PR TITLE
[REVIEW][skip-ci] Updating documentation and fixing naming consistentcy

### LINKED ISSUE #863: [ENH] Update the docs to include conding and PR standards
The team using a number of tags inside of code and Pull Requests to identify technical debt and other items.  Document the coding and tagging standard

### LINKED ISSUE #929: [BUG] documentation error
**Describe the bug**

https://github.com/rapidsai/cugraph/blob/branch-0.15/cpp/include/graph.hpp#L486
https://github.com/rapidsai/cugraph/blob/branch-0.15/cpp/include/graph.hpp#L539

offsets & indices appear in the documentation of GraphCSR & GraphCSC constructors, but they do not appear in actual declaration.

Also, ordering of the parameters in the documentation and declaration do not match.

### PR DESCRIPTION
**Pull request by BradReesWork**
closes #929 
closes #863

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by BradReesWork** [comment]
rerun tests

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -74,11 +74,14 @@ If you need more context on a particular issue, please ask.
 Remember, if you are unsure about anything, don't hesitate to comment on issues
 and ask for clarifications!
 
+**The _TODO_** comment<pr>
+
+Use the _TODO_ comment to capture technical debt.  It should not be used to flag areas that need to be fixed.  FixMe blocks need to be cleaned up before code is submitted.
```
Comment (ChuckHastings) [inline CONTRIBUTING.md:79]: I thought we had defined these the other way when we discussed this a few months ago (TODO should be cleaned up before code is submitted, FIXME to define technical debt).  But I may be remembering wrong.

File: PRTAGS.md [hunk 2]
```diff
@@ -9,4 +9,4 @@ PR = Pull Request
 | skip-ci    | _Do Not Run CI_ - This flag prevents CI from being run.  It is good practice to include this with the **WIP** tag since code is typically not at a point where it will pass CI.  |
 | skip ci    | same as above                                          |
 | API-REVIEW | This tag request a code review just of the API portion of the code - This is benificial to ensure that all required arguments are captured.  Doing this early can save from having to refactor later. |
```
Comment (ChuckHastings) [inline PRTAGS.md:11]: spelling error:   beneficial

File: README.md [hunk 3]
```diff
@@ -79,26 +80,25 @@ for i in range(len(df_page)):
...
 
 cuGraph is constantly being updated and improved. Please see the [Transition Guide](TRANSITIONGUIDE.md) if errors are encountered with newer versions
 
 ## Graph Sizes and GPU Memory Size
 The amount of memory required is dependent on the graph structure and the analytics being executed.  As a simple rule of thumb, the amount of GPU memory should be about twice the size of the data size.  That gives overhead for the CSV reader and other transform functions.  There are ways around the rule but using smaller data chunks.
 
-
 |       Size        | Recommended GPU Memory |
 |-------------------|------------------------|
 | 500 million edges |  32GB                  |
```
Comment (ChuckHastings) [inline README.md:98]: Formatting...  with or without space?  "32GB" or "32 GB"?

File: CONTRIBUTING.md (hunk 1 above)
Comment (BradReesWork) [reply CONTRIBUTING.md:79]: ooops.  Fixing that now

File: PRTAGS.md (hunk 2 above)
Comment (BradReesWork) [reply PRTAGS.md]: fixed

File: README.md (hunk 3 above)
Comment (BradReesWork) [reply README.md]: fixed.  both use a space

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -74,11 +74,14 @@ If you need more context on a particular issue, please ask.
 Remember, if you are unsure about anything, don't hesitate to comment on issues
 and ask for clarifications!
 
+**The _FIXME_** comment<pr>
+
+Use the _FIXME_ comment to capture technical debt.  It should not be used to flag flag bugs since those need to be cleaned up before code is submitted.
```
Comment (seunghwak) [inline CONTRIBUTING.md:79]: "flag flag" => flag

File: cpp/include/graph.hpp [hunk 5]
```diff
@@ -212,9 +215,12 @@ class GraphCompressedSparseBaseView : public GraphViewBase<VT, ET, WT> {
...
    * @param  number_of_edges       The number of edges in the graph
    */
-  GraphCompressedSparseBaseView(
-    ET *offsets_, VT *indices_, WT *edge_data_, VT number_of_vertices_, ET number_of_edges_)
-    : GraphViewBase<VT, ET, WT>(edge_data_, number_of_vertices_, number_of_edges_),
+  GraphCompressedSparseBaseView(edge_t *offsets_,
+                                vertex_t *indices_,
+                                weight_t *edge_data_,
+                                vertex_t number_of_vertices_,
+                                edge_t number_of_edges_)
```
Comment (seunghwak) [inline cpp/include/graph.hpp:222]: variable names with trailing "_" is for class private member variables (RAFT clang-tidy rules).

File: cpp/include/graph.hpp [hunk 6]
```diff
@@ -257,9 +266,12 @@ class GraphCSRView : public GraphCompressedSparseBaseView<VT, ET, WT> {
...
    * @param  number_of_edges       The number of edges in the graph
    */
-  GraphCSRView(
-    ET *offsets_, VT *indices_, WT *edge_data_, VT number_of_vertices_, ET number_of_edges_)
-    : GraphCompressedSparseBaseView<VT, ET, WT>(
+  GraphCSRView(edge_t *offsets_,
+               vertex_t *indices_,
+               weight_t *edge_data_,
+               vertex_t number_of_vertices_,
+               edge_t number_of_edges_)
```
Comment (seunghwak) [inline cpp/include/graph.hpp:273]: variable names with trailing "_" is for class private member variables (RAFT clang-tidy rules).

File: cpp/include/graph.hpp [hunk 7]
```diff
@@ -522,117 +524,101 @@ class GraphCompressedSparseBase {
...
    * @param  number_of_edges       The number of edges in the graph
+   * @param  has_data              Wiether or not the class has data, default = False
+   * @param  stream                Specify the cudaStream, default = null
+   * @param mr                     Specify the memory resource
    */
-  GraphCSR(VT number_of_vertices_,
-           ET number_of_edges_,
+  GraphCSR(vertex_t number_of_vertices_,
+           edge_t number_of_edges_,
            bool has_data_                      = false,
```
Comment (seunghwak) [inline cpp/include/graph.hpp:550]: variable names with trailing "_" is for class private member variables (RAFT clang-tidy rules).

File: cpp/include/graph.hpp [hunk 8]
```diff
@@ -522,117 +524,101 @@ class GraphCompressedSparseBase {
...
    * @param  number_of_edges       The number of edges in the graph
+   * @param  has_data              Wiether or not the class has data, default = False
+   * @param  stream                Specify the cudaStream, default = null
+   * @param mr                     Specify the memory resource
    */
-  GraphCSC(VT number_of_vertices_,
-           ET number_of_edges_,
+  GraphCSC(vertex_t number_of_vertices_,
+           edge_t number_of_edges_,
            bool has_data_                      = false,
```
Comment (seunghwak) [inline cpp/include/graph.hpp:601]: variable names with trailing "_" is for class private member variables (RAFT clang-tidy rules).

File: cpp/include/graph.hpp (hunk 5 above)
Comment (BradReesWork) [reply cpp/include/graph.hpp:222]: oops.  missed that mix.

File: CONTRIBUTING.md (hunk 4 above)
Comment (BradReesWork) [reply CONTRIBUTING.md]: fixed

File: cpp/include/graph.hpp (hunk 6 above)
Comment (BradReesWork) [reply cpp/include/graph.hpp]: fixed

File: cpp/include/graph.hpp (hunk 8 above)
Comment (BradReesWork) [reply cpp/include/graph.hpp]: fixed

**Comment by BradReesWork** [comment]
rerun tests

**Comment by BradReesWork** [comment]
switching to skip-ci to resolve conflict without a rebuild

**Pull request merged**